src/misc/credentials/CredentialsProviderFactory.js

```diff
@@ -19,14 +19,12 @@ export function usingKeychainAuthentication(): boolean {
  * @param deviceEncryptionFacade
  * @param nativeApp
  */
-export function createCredentialsProvider(deviceEncryptionFacade: DeviceEncryptionFacade, nativeApp: NativeInterface): ICredentialsProvider {
-	if (usingKeychainAuthentication()) {
+export function createCredentialsProvider(deviceEncryptionFacade: DeviceEncryptionFacade, nativeApp: ?NativeInterface): ICredentialsProvider {
+	if (usingKeychainAuthentication() && nativeApp) {
```

Comment:
no
if we are using keychain, the app *must* be there, I don't want to *not* use it accidentally because we forgot to pass it in. Assert instead pls.